src/figuresClasses.ts

```diff
@@ -2,10 +1,98 @@
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  color: Color;
+  shape: Shape;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0) {
+      throw new Error('Side "a" cannot be less than or equal to zero');
+    }
+
+    if (b <= 0) {
+      throw new Error('Side "b" cannot be less than or equal to zero');
+    }
+
+    if (c <= 0) {
+      throw new Error('Side "c" cannot be less than or equal to zero');
+    }
```

Comment:
[CODE STYLE] - While you are throwing errors for invalid input, the error messages could be a bit more descriptive. For instance, instead of 'Side "a" cannot be less than or equal to zero', you could say 'The length of side "a" is ${a}, which cannot be zero or less'. This gives the user more information about what went wrong.